### ISSUE TITLE
Batch mode with returnPromise=true

### ISSUE THREAD

**Issue by Amphiluke** [clean-css/clean-css#1186]
Hello,

I’m trying to implement a batched minification of an array of hashes as stated in the [docs](https://github.com/clean-css/clean-css/blob/9ae9089e278598e28fba334978ff5130fc98e8aa/README.md#how-to-process-multiple-files-without-concatenating-them-into-one-output-file). Everything works fine in synchronous mode. But if I set the option [`returnPromise: true`](https://github.com/clean-css/clean-css/blob/9ae9089e278598e28fba334978ff5130fc98e8aa/README.md#promise-interface), the returned promise always gets rejected in spite of the minification seems to complete successfully behind the scenes.

```javascript
let CleanCSS = require('clean-css');

(async () => {
    try {
        let cleanCSS = new CleanCSS({
            batch: true,
            returnPromise: true,
        });
        let outputs = await cleanCSS.minify([
            {'path/to/file/one': {styles: 'html { color: #000000; }'}},
            {'path/to/file/two': {styles: 'body { background: #ffffff; }'}}
        ]);
        console.log('OK', outputs);
    } catch (e) {
        console.log('error', e); // <-- always logs: "error" [null, null]
    }
})();
```

Probably, this happens because there is no check for `innerErrors === null` in the [function `whenHashBatchDone`](https://github.com/clean-css/clean-css/blob/9ae9089e278598e28fba334978ff5130fc98e8aa/lib/clean.js#L104).

----

### Precheck
 

- [x] Do a quick search and make sure a bug has not yet been reported;
- [x] do a quick check if the bug still exists in the latest patch version;
- [x] finally, be nice and have fun!

### Environment

* clean-css version: 5.2.0
* node.js version: 14.17.6
* operating system: any

### Configuration options

```js
var CleanCSS = require('clean-css');
new CleanCSS({
    batch: true,
    returnPromise: true,
});
```

### Demo link for the issue reproduction

[clean-css-batch-promise](https://runkit.com/amphiluke/clean-css-batch-promise)

**Comment by jakubpawlowicz** [comment]
Thanks @Amphiluke for pinpointing the exact reason - can you try again with fa23d2b687407192777a08f318d5e0b3db1e475e ? I can release it if it works for you.

**Comment by Amphiluke** [comment]
Hi! Checked out the commit you mentioned, and now it works perfectly for me. Thank you for the quick fix. :)

**Comment by jakubpawlowicz** [comment]
Cool, I'll push it soon as 5.2.1.

**Comment by jakubpawlowicz** [comment]
And it's released in 5.2.1. Cheers :beer:
